packages/node/test/local/assignment/assignment-filter.test.ts

```diff
@@ -0,0 +1,194 @@
+import { Assignment } from 'src/assignment/assignment';
+import { InMemoryAssignmentFilter } from 'src/assignment/assignment-filter';
+import { ExperimentUser } from 'src/types/user';
+import { sleep } from 'src/util/time';
+import { MockAssignmentFilter } from 'test/local/util/mockAssignmentFilter';
+
+test('filter - single assignment', async () => {
```

Comment:
I'm skimming the tests, just note:
**Please add more tests if you don't understand some part of the code and want to make sure it's doing what you expect**

You will have to fix all the bugs that eventually make it into production :)